piu/cli.py

```diff
@@ -168,36 +191,38 @@ def _request_access(even_url, cacert, username, hostname, reason, remote_host, l
     )
     if r.status_code == 200:
         click.secho(r.text, fg="green", bold=True)
-        ssh_command = ""
-        if remote_host:
-            ssh_command = "ssh -o StrictHostKeyChecking=no {username}@{remote_host}".format(**vars())
-            if tunnel:
-                ports = tunnel.split(":")
-                ssh_command = "-L {local_port}:{remote_host}:{remote_port}".format(
-                    local_port=ports[0], remote_host=remote_host, remote_port=ports[1]
-                )
-        command = "ssh -tA {username}@{hostname} {ssh_command}".format(
-            username=username, hostname=hostname, ssh_command=ssh_command
-        )
-        if connect or tunnel:
-            subprocess.call(command.split())
-
-        if not ssh_keys_added():
-            warning("No SSH identities found. Please add one using ssh-add, for example:")
-            warning("ssh-add ~/.ssh/id_rsa")
-
-        click.secho("You can access your server with the following command:")
-        click.secho(command)
-
-        if clip:
-            click.secho('\nOr just check your clipboard and run ctrl/command + v (requires package "xclip" on Linux)')
-            if pyperclip is not None:
-                pyperclip.copy(command)
+        ssh_connection(clip, connect, hostname, remote_host, tunnel, username, username)
     else:
         click.secho(
             "Server returned status {code}: {text}".format(code=r.status_code, text=r.text), fg="red", bold=True
         )
-    return r.status_code
+    return r.status_code == 200
+
+
+def ssh_connection(clip, connect, hostname, remote_host, tunnel, odd_user, remote_user):
+    if tunnel:
+        ports = tunnel.split(":")
+        ssh_command = "ssh {odd_user}@{hostname} -L {local_port}:{remote_host}:{remote_port}".format(
+            odd_user=odd_user, hostname=hostname, local_port=ports[0], remote_host=remote_host, remote_port=ports[1]
+        )
+    else:
+        if remote_host:
+            ssh_command = "ssh -J {odd_user}@{odd_host} {remote_user}@{remote_host}".format(
+                odd_user=odd_user, odd_host=hostname, remote_user=remote_user, remote_host=remote_host
+            )
+        else:
+            ssh_command = "ssh {odd_user}@{odd_host}".format(odd_user=odd_user, odd_host=hostname)
+    if connect or tunnel:
+        subprocess.call(ssh_command.split())
```

Comment:
We should be more strict about the arguments and construct the command line without relying on `split`. It's more verbose, but then accidental spaces in the arguments don't cause this thing to break. It also makes it so that future changes to e.g. add a command to execute would not break with spaces.

Reply:
Although since this was a leftover from the previous code we should probably fix it separately.

Reply:
Since the user input is already sanitized the chance of extra spaces is none. So I would not fix it now anyway.